File: 30days/day25-running-time-and-complexity/Solution.js
// PROBLEM: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem

function processData(input) {
    //Enter your code here
    const numbers = input.split('\n').splice(1).map(Number);

    numbers.forEach((n) => {
        process.stdout.write(isPrime(n) ? 'Prime\n' : 'Not prime\n');
    });
} 

function isPrime(number) {
    if (number <= 1){
        return false;
    }
    for (let i = 2; i*i <= number; i++) {
        if (number % i === 0) {
            return false;
        }
    }
    return true;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: 30days/day25-running-time-and-complexity/Solution.cs
// PROBLEM: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem

using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int numbers = Convert.ToInt32(Console.ReadLine());

        for (int i = 0; i < numbers; i++) {
            int number = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine(IsPrime(number) ? "Prime" : "Not prime");
        }
    }

    private static bool IsPrime(int number) {
        if (number <= 1){
            return false;
        }
        for (int i = 2; i*i <= number; i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }
}